sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -360,20 +358,22 @@ object ExecutionPlanCaptureCallback {
     executedPlan.expressions.exists(didFallBack(_, fallbackCpuClass))
   }
 
-  private def containsExpression(exp: Expression, className: String): Boolean = {
-    PlanUtils.getBaseNameFromClass(exp.getClass.getName) == className ||
-        exp.children.exists(containsExpression(_, className))
-  }
-
-  private def containsPlan(plan: SparkPlan, className: String): Boolean = {
-    val p = ExecutionPlanCaptureCallback.extractExecutedPlan(Some(plan)) match {
-      case p: QueryStageExec => p.plan
-      case p => p
-    }
-    PlanUtils.sameClass(p, className) ||
-        p.expressions.exists(containsExpression(_, className)) ||
-        p.children.exists(containsPlan(_, className))
-  }
+  private def containsExpression(exp: Expression, className: String): Boolean = exp.find {
+    case e if PlanUtils.getBaseNameFromClass(e.getClass.getName) == className => true
+    case e: ExecSubqueryExpression => containsPlan(e.plan, className)
+    case _ => false
+  }.nonEmpty
+
+  private def containsPlan(plan: SparkPlan, className: String): Boolean = plan.find {
+    case p if PlanUtils.sameClass(p, className) =>
+      true
+    case p: AdaptiveSparkPlanExec =>
+      containsPlan(p.executedPlan, className)
+    case p: QueryStageExec =>
+      containsPlan(p.plan, className)
+    case p =>
+      p.expressions.exists(containsExpression(_, className))
```

Comment:
The name `containsPlan` implies it will search the plan tree, but this is only looking at the expressions of the current plan node, ignoring any children plan nodes.  Looks like child plan nodes will only be searched if the node happens to be `AdaptiveSparkPlanExec` or `QueryStageExec`.

Reply:
I think the `plan.find` will take care of the search of the plan tree.